### ISSUE TITLE
document REPL `ast_transforms`

### ISSUE THREAD

**Issue by stevengj** [JuliaLang/julia#37047]
Julia 1.5 includes a REPL `ast_transforms` variable initially added for the softscope transformation (#33864) but expanded for use by other packages  (#34626, #34778),  that is starting to be used by other packages.  It is currently being used by Revise (https://github.com/timholy/Revise.jl/pull/425), support will probably be added soon in IJulia (for the `REPL.repl_ast_transforms` defaults), and it might be used in the future by things like OhMyREPL.

We might consider documenting it somewhere.

**Comment by rfourquet** [comment]
> Maybe adding a slightly higher-level API, e.g.

> `REPL.current_ast_transforms() = isdefined(Base, :active_repl_backend) ? Base.active_repl_backend.ast_transforms : REPL.repl_ast_transforms`

FWIW, this is almost exactly what `SafeREPL` [uses](https://github.com/rfourquet/SafeREPL.jl/blob/11eb244698a731e2eb51d2beed6b68ecb2e401b2/src/SafeREPL.jl#L22).

**Comment by stevengj** [comment]
Note that there is the undocumented function `REPL.Numbered.__current_ast_transforms()`: https://github.com/JuliaLang/julia/blob/39f2ad1e941fd092b906d60e8d23c004e8ee5b7e/stdlib/REPL/src/REPL.jl#L1786

**Comment by tpapp** [comment]
Reviving this issue after a while: `REPL.repl_ast_transforms` has been around for years now with the same API, so would it make sense to document *as is*? Namely,

1. it is a `Vector{Any}`, just use the relevant `push!` etc API to modify it
2. each element does an AST transform, they are applied in the given order,
3. (AFAIK) it is only read when the REPL is *started*, so see either use `atreplinit`, or `Base.active_repl_backend.ast_transforms` if you want to extend your already running REPL

**Comment by stevengj** [comment]
Right, `REPL.repl_ast_transforms` has no effect if an active REPL has already been instantiated.

**Comment by tpapp** [comment]
Yes, but that knowledge is currently implicit in the code; all I am suggesting is documenting the 3 points above. Possibly as experimental API that may change with minor versions without prior notice. If people are open to this I can make a PR.
